pkg/ottl/ottlfuncs/README.md

```diff
@@ -1208,6 +1209,30 @@ Examples:
 
 **Note:** According to the National Institute of Standards and Technology (NIST), SHA256 is no longer a recommended hash function. It should be avoided except when required for compatibility. New uses should prefer FNV whenever possible.
 
+### Sort
+
+`Sort(target, Optional[order])`
+
+The `Sort` Converter sorts the `target` array in ascending or descending order.
+
+`target` is a `pcommon.Slice` type field. `order` is a string that must be one of `asc` or `desc`. The default `order` is `asc`.
+
+If elements in `target` are
```

Comment:
What are your thoughts on permanently converting the types of values in heterogeneous arrays vs. just converting the types for comparison? I think it would be fairly straightforward to only convert the types within the sort function and leave the actual type in the array. I can see arguments for both sides here, but want to make sure we're considering our options.

Reply:
I see changing the types permanently and sorting a list as two distinct intentions. Both are reasonable actions in data processing. Here, we provide better flexibility to users. If they want to change the types permanently, they can simply overwrite the attributes by using `set()`. Otherwise, they will still have the original array on their hands

Reply:
That makes sense, thanks. If we see changing the types and sorting as separate actions, then shouldn't we keep the original types of the array elements when sorting? Right now we do:

```
[true, 1, 2.3, "str"] --> ["1", "2.3", "str", "true"]
```

When I think it would be possible to do:

```
[true, 1, 2.3, "str"] --> [1, 2.3, "str", true]
```

Reply:
I see your point. Yes, preserving the types may be more intuitive. I thought keeping the sort result as a string to make it clear that the comparison is done as a string. It sounds good to maintain the original type.